Shared/Samples/Create KML multi-track/CreateKMLMultiTrackView.swift

```diff
@@ -0,0 +1,185 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct CreateKMLMultiTrackView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The asynchronous action currently being run.
+    @State private var asyncAction: AsyncAction? = .startNavigation
+    
+    /// The text shown in the status bar. This describes the current state of the sample.
+    @State private var statusText = ""
+    
+    /// The KML multi-track loaded from the KMZ file.
+    @State private var multiTrack: KMLMultiTrack?
+    
+    /// A Boolean value indicating whether the recenter button is enabled.
+    @State private var recenterIsEnabled = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        MapViewReader { mapViewProxy in
+            MapView(
+                map: model.map,
+                graphicsOverlays: [model.trackElementGraphicsOverlay, model.trackGraphicsOverlay]
+            )
+            .locationDisplay(model.locationDisplay)
+            .overlay(alignment: .top) {
+                Text(statusText)
+                    .multilineTextAlignment(.center)
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .padding(8)
+                    .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
+            }
+            .toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    if let multiTrack {
+                        Button("Delete", systemImage: "trash", role: .destructive) {
+                            asyncAction = .reset
+                        }
+                        
+                        Spacer()
+                        
+                        TrackPicker(multiTrack: multiTrack, mapViewProxy: mapViewProxy)
+                    } else {
+                        Button("Recenter", systemImage: "location.north.circle") {
+                            model.locationDisplay.autoPanMode = .navigation
+                        }
+                        .disabled(!recenterIsEnabled)
+                        
+                        Spacer()
+                        
+                        Button(asyncAction == .recordTrack ? "Stop Recording" : "Record Track") {
+                            if asyncAction == .recordTrack {
+                                asyncAction = nil
+                                model.addTrack()
+                            } else {
+                                asyncAction = .recordTrack
+                            }
+                        }
```

Comment:
For a toggle action like this, consider using a toggle button for an added visual cue:

```swift
Toggle(
    asyncAction == .recordTrack ? "Stop Recording" : "Record Track",
    isOn: .init {
        asyncAction == .recordTrack
    } set: { newValue in
        asyncAction = newValue ? .recordTrack : nil
    }
)
```

![Screen Recording 2025-02-28 at 11 03 39 AM](https://github.com/user-attachments/assets/e8119fda-0c46-46eb-bc7d-ec8a9d4c6b2c)

However, you'll need to call `addTrack()` some other way. You could do that in the task modifier as a result of the state change.